swarm/src/lib.rs

```diff
@@ -973,11 +977,11 @@ where
                 //
                 // The translation is transport-specific. See [`Transport::address_translation`].
                 let translated_addresses = {
-                    let transport = self.listeners.transport();
                     let mut addrs: Vec<_> = self
-                        .listeners
-                        .listen_addrs()
-                        .filter_map(move |server| transport.address_translation(server, &address))
+                        .listened_addrs
+                        .values()
+                        .flatten()
+                        .filter_map(|server| self.transport.address_translation(server, &address))
```

Comment:
Unrelated to this PR but `translate_address` might be a more natural name for this function.